src/server/scripts/BrokenIsles/NeltharionsLair/boss_rokmora.cpp

```diff
@@ -0,0 +1,569 @@
+    SPELL_RUPTURING_SKITTER              = 215929,
+    SPELL_SHATTER                        = 188114,
+    SPELL_SHATTER_KILL_SKITTERS          = 200247,
+
+    SPELL_ROKMORA_COMBAT_CONVO_1         = 198122,  // Rok shatter!
+    SPELL_ROKMORA_COMBAT_CONVO_2         = 198125,  // Rok stronger! (unused?)
+    SPELL_ROKMORA_COMBAT_CONVO_3         = 198135,  // Tiny piddle, big puddle.
+    SPELL_ROKMORA_COMBAT_CONVO_4         = 198136   // Nothing beat Rok!
+};
+
+enum RokmoraEvents
+{
+    EVENT_SHATTER = 1,
+    EVENT_RAZOR_SHARDS,
+    EVENT_CRYSTALLINE_GROUND,
+    EVENT_CHECK_ENERGY,
+
+    EVENT_ROKMORA_INTRO_MOVE
+};
+
+enum RokmoraTexts
+{
+    SAY_AGGRO                      = 0,
+    SAY_CRYSTALLINE_GROUND_WARNING = 1,
+    SAY_RAZOR_SHARDS_WARNING       = 2,
+    SAY_DIED                       = 3
+};
+
+enum RokmoraConversation
+{
+    POINT_INTRO                 = 0,
+    CONVO_ACTOR_NAVARROGG       = 52530,
+    CONVO_ULAROGG_INTRO_LINE_03 = 3992,
+
+    CONVO_ACTOR_ID_ULAROGG      = 0,
+    CONVO_ACTOR_ID_NAVARROGG    = 1,
+
+    PATH_ULAROGG                = 10530000,
+    PATH_NAVARROGG              = 10070000,
+
+    WAYPOINT_DESPAWN            = 7
+};
+
+enum RokmoraMisc
+{
+    NPC_NAVARROGG      = 100700,
+    NPC_ULAROGG        = 105300,
+    NPC_SKITTER        = 97720
+};
+
+constexpr Position NavarroggIntroPosition = { 2959.8716f, 1390.5312f, 53.52835f, 2.721162f };
+constexpr Position NavarroggIntroPosition2 = { 2917.3196f, 1402.2882f, -2.405084f };
+constexpr Position NavarroggJumpIntroPosition = { 2930.9895f, 1395.7848f, -2.641485f, 2.721162f };
+
+// 91003 - Rokmora
+struct boss_rokmora : public BossAI
+{
+    boss_rokmora(Creature* creature) : BossAI(creature, DATA_ROKMORA) { }
+
+    void JustAppeared() override
+    {
+        if (instance->GetData(DATA_ROKMORA_INTRO_STATE) == DONE)
+            me->SetImmuneToPC(false);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        _JustDied();
+        Talk(SAY_DIED);
+    }
+
+    void KilledUnit(Unit* victim) override
+    {
+        if (victim->GetEntry() == NPC_SKITTER)
+        {
+            uint32 spellId = RAND(SPELL_ROKMORA_COMBAT_CONVO_3, SPELL_ROKMORA_COMBAT_CONVO_4);
+            DoCastSelf(spellId);
+        }
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+
+        summons.DespawnAll();
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        Talk(SAY_AGGRO);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me, 1);
+
+        DoCastSelf(SPELL_GAIN_ENERGY);
+
+        events.ScheduleEvent(EVENT_RAZOR_SHARDS, 29100ms);
+        events.ScheduleEvent(EVENT_CHECK_ENERGY, 500ms);
+
+        if (IsHeroicOrHigher()) // Casted once
```

Comment:
it used to be casted quite a few times, maybe its bugged on retail atm that it only casts it once?
Check DBM/BigWigs for timers after